UPGRADING.md

```diff
@@ -0,0 +1,110 @@
+
+If you experience issues or have questions, please file an [issue](https://github.com/googleapis/python-monitoring-dashboards/issues).
+
+## Supported Python Versions
+
+> **WARNING**: Breaking change
+The 2.0.0 release requires Python 3.6+.
+
+## Method Calls
+
+> **WARNING**: Breaking change
+Methods expect request objects. We provide a script that will convert most common use cases.
+* Install the library
+
+```py
+python3 -m pip install google-cloud-monitoring-dashboards
+```
+
+* The scripts `fixup_dashboard_v1_keywords.py` shipped with the library. It expects
+an input directory (with the code to convert) and an empty destination directory.
+
+```sh
+$ fixup_dashboard_v1_keywords.py --input-directory .samples/ --output-directory samples/
+```
+
+**Before:**
+```py
+from google.cloud.monitoring_dashboard import v1
+client = v1.DashboardsServiceClient()
+# TODO: Initialize `parent`:
+parent = ''
+# TODO: Initialize `dashboard`:
+dashboard = {}
+response = client.create_dashboard(parent, dashboard)
+```
+
+
+**After:**
+```py
+from google.cloud.monitoring_dashboard import v1
+client = v1.DashboardsServiceClient()
+response = client.create_dashboard(request={"parent": "''", "dashboard": "{}"})
+```
+
+### More Details
+
+In `google-cloud-monitoring-dashboards<2.0.0`, parameters required by the API were positional parameters and optional parameters were keyword parameters.
+
+**Before:**
+```py
+    def create_dashboard(
+        self,
+        parent,
+        dashboard,
+        retry=google.api_core.gapic_v1.method.DEFAULT,
+        timeout=google.api_core.gapic_v1.method.DEFAULT,
+        metadata=None,
+    ):
+```
+
+In the 2.0.0 release, all methods have a single positional parameter `request`. Method docstrings indicate whether a parameter is required or optional.
+
+Some methods have additional keyword only parameters. The available parameters depend on the [`google.api.method_signature` annotation] specified by the API producer.
+
+
+**After:**
+```py
+    def create_dashboard(
+        self,
+        request: dashboards_service.CreateDashboardRequest = None,
+        *,
+        retry: retries.Retry = gapic_v1.method.DEFAULT,
+        timeout: float = None,
+        metadata: Sequence[Tuple[str, str]] = (),
+    ) -> dashboard.Dashboard:
+```
+
+> **NOTE:** The `request` parameter and flattened keyword parameters for the API are mutually exclusive.
+> Passing both will result in an error.
+Both of these calls are valid:
+
+```py
+response = client.create_dashboard(
+    request={
+        "parent": parent,
+        "dashboard": dashboard,
+    }
+)
+```
+
+```py
+response = client.create_dashboard(
+    parent=parent,
+    dashboard=dashboard,
+)
+```
+
+This call is invalid because it mixes `request` with a keyword argument `dashboard`. Executing this code will result in an error.
+
+```py
+response = client.create_dashboard(
+    request={
+        "parent": parent,
+    },
+    dashboard=dashboard
+)
+```
```

Comment:
I remember there is a section after this about `Enums and Types`.

Reply:
Yes, while I searched `enums` in code base, and seems we don't use enums anywhere, adding it to the upgrading doc might confuse users, WDYT? Thanks!

Reply:
Given that the present directory layout is borked I think you're right that it'll be more confusing to have that section.

Reply:
yes that makes sense.